Sources/MapboxDirections/RouteStep.swift

```diff
@@ -582,27 +675,24 @@ open class RouteStep: Codable {
     
     init(from decoder: Decoder, administrativeRegions: [AdministrativeRegion]?) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        let maneuver = try container.nestedContainer(keyedBy: ManeuverCodingKeys.self, forKey: .maneuver)
         
-        maneuverLocation = try maneuver.decode(LocationCoordinate2DCodable.self, forKey: .location).decodedCoordinates
-        maneuverType = (try? maneuver.decode(ManeuverType.self, forKey: .type)) ?? .default
-        maneuverDirection = try maneuver.decodeIfPresent(ManeuverDirection.self, forKey: .direction)
-        exitIndex = try maneuver.decodeIfPresent(Int.self, forKey: .exitIndex)
-
-        initialHeading = try maneuver.decodeIfPresent(Turf.LocationDirection.self, forKey: .initialHeading)
-        finalHeading = try maneuver.decodeIfPresent(Turf.LocationDirection.self, forKey: .finalHeading)
+        let maneuver = try container.decode(Maneuver.self, forKey: .maneuver)
+        
+        maneuverLocation = maneuver.maneuverLocation
+        maneuverType = maneuver.maneuverType
+        maneuverDirection = maneuver.maneuverDirection
+        exitIndex = maneuver.exitIndex
+        initialHeading = maneuver.initialHeading
+        finalHeading = maneuver.finalHeading
+        instructions = maneuver.instructions
+        maneuverForeignMembers = maneuver.foreignMembers
         
         if let polyLineString = try container.decodeIfPresent(PolyLineString.self, forKey: .shape) {
             shape = try LineString(polyLineString: polyLineString)
         } else {
             shape = nil
         }
         
-        if let instruction = try? maneuver.decode(String.self, forKey: .instruction) {
-            instructions = instruction
-        } else {
-            instructions = "\(maneuverType) \(maneuverDirection?.rawValue ?? "")"
```

Comment:
FYI, this was a fallback for custom OSRM endpoints that had not integrated OSRM Text Instructions. We primarily used it for debugging purposes, since it was easier to fashion a test fixture using OSRM’s demo page than using the Directions API at the time. These days, any OSRM endpoint would be expected to integrate OSRMTI.